### DISCUSSION TITLE
name of enum can be omitted

### DISCUSSION THREAD

**Discussion by martindrlik** [dotnet/csharplang#5669] [Language Ideas] [answered]
Hello guys, in order to have shorter code I would like to omit enum name. If the "enum type" can be deduced, only dot can be used to access enum item. Please see following example:
```
enum Example {
    Foo,
    Bar,
    Baz,
}

void DoSomething(Example example) { ... }
DoSomething(.Foo);

var example = Example.Bar;
Example ex = .Baz;

switch (example) {
    case .Foo: ...
    case .Bar: ...
    case .Baz: ...
}
```

What do you think?

**Comment by 333fred** [comment, marked as answer]
This is championed in https://github.com/dotnet/csharplang/issues/2926.
